core/src/main/java/edu/stanford/futuredata/macrobase/pipeline/BasicBatchPipeline.java

```diff
@@ -24,45 +26,48 @@
 public class BasicBatchPipeline implements Pipeline {
     Logger log = LoggerFactory.getLogger(Pipeline.class);
 
-    private String inputURI = null;
+    private final PipelineConfig conf;
 
-    private String classifierType;
-    private String metric;
-    private double cutoff;
-    private boolean pctileHigh;
-    private boolean pctileLow;
-
-    private String summarizerType;
-    private List<String> attributes;
-    private String ratioMetric;
-    private double minSupport;
-    private double minRiskRatio;
+    // PipelineConfig params applicable to all classifiers
+    private final String metric;
+    private final String inputURI;
```

Comment:
I agree it's a good idea to move commonly used config params to a common place, however one thing I don't like about this approach is that now all of the conf params a pipeline needs are spread out in random places. I would much rather have them gathered in a way where it's easy to find all of the params one might need for a given pipeline.

What do we really gain from moving the conf parameter lookups to the clasifier / summarizer initialization section? Can we get those benefits while still making the complete set of parameters easy to find?